src/containers/ChartInfo/index.js

```diff
@@ -42,14 +42,27 @@ const PpmResizedChart = windowSize(PpmChart);
 
 export class ChartInfo extends Component {
   state = {
-    rangeType: FIVE_YEAR, // Intitial date range query type
+    rangeType:
+      this.props.match && this.props.match.params.time
+        ? this.props.match.params.time
+        : FIVE_YEAR, // Intitial date range query type
   };
 
   componentDidMount() {
-    // query for current ppm
-    this.props.fetchCurrentPpms();
-    // Query for five years
-    this.queryApi(this.state.rangeType);
+    // Wait for the total number of records to load
+    // before requesting ALL records from the API.
+    if (this.state.rangeType === ALL) {
+      // query for current ppm and wait for limit
+      Promise.resolve(this.props.fetchCurrentPpms()).then(() => {
```

Comment:
`fetchCurrentPpms` already returns a promise, so I am wondering do we need the call to `fetchCurrentPpms` wrapped in a `Promise.resolve` ?